src/parse_execution.rs

```diff
@@ -567,6 +568,19 @@ impl<'a> ParseExecutionContext {
                 "The expanded command was empty."
             );
         }
+        // Complain if we've expanded to a subcommand keyword like "command" or "if".
+        // This is an attempt to defeat function resolution.
+        //
+        // Make an exception for "time" because that is frequently used as a command and does fundamentally the same thing.
+        if parser_keywords_is_subcommand(out_cmd) && !unexp_cmd.starts_with(out_cmd.chars()) && out_cmd != L!("time") {
```

Comment:
in addition to `time` we could allow `not` which ships with LLVM though I think we're fine either way.
LGTM other than my above comment.

Reply:
I've been thinking of removing the `time` exception, actually, because it doesn't work:

```fish
set -l time time
$time sleep 5
```

Gives you the `time` man page instead of running the `time` command.

Reply:
since it never worked that sounds fine, we can make it work later if anyone ever cares